### PR TITLE
Refactor/abstract module route

### PR DESCRIPTION
**Pull request by billybonks**
While i was reading through some code,
-  I felt it was easier to read if i removed `bind(this)`.
- While i was at it i used es6 functions.
- Added some deconstruction for used ember packages.
- I refactored `abstract-index-route.js` to make it a bit more readable.

### CODE REVIEW

File: app/routes/abstract-module-route.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
-import AuthenticatedRouteMixin from 'ember-simple-auth/mixins/authenticated-route-mixin';
 import Ember from 'ember';
+import AuthenticatedRouteMixin from 'ember-simple-auth/mixins/authenticated-route-mixin';
```
Comment (jkleinsc) [inline app/routes/abstract-module-route.js]: @billybonks Why move  `import Ember from 'ember';` to the top?

Comment (billybonks) [inline app/routes/abstract-module-route.js]: I find it more pleasing to the eye when i have ember on top, of course not a good reason. do you not like it there ? @taras do you have an opinion ?

Comment (jkleinsc) [inline app/routes/abstract-module-route.js]: @billybonks I've tried to keep the imports alphabetical.  I'm not sure I see a reason to put it at the top.  I understand and support the other changes, but moving the Ember import feels superfluous.  If I recall correctly I think I heard that at some point the Ember import goes away, so I suppose you could make a small argument to put it at the top for easy removal later?

Comment (jkleinsc) [inline app/routes/abstract-module-route.js]: @billybonks I took a quick look at our code and it looks like its about half and half as to the Ember import being at the top of the file or alphabetically, so there really isn't a consistent standard.  I'm fine moving along this PR as is.

**Comment by billybonks** [comment]
ok, i am happy to change it as well, alphabetical makes sense, i'll write
all future code as per the style of the repo :). but yes it was a
superfluous change

**Comment by jkleinsc** [comment]
@billybonks just want to check to see if you are still working on this one.

**Comment by billybonks** [comment]
Hey ill do a rebase this weekend. sorry for the delay things have been hectic this side.

**Comment by jkleinsc** [comment]
@billybonks no worries... just wanted to verify you were still working on this.  Thanks!

**Comment by billybonks** [comment]
closing this

**Pull request closed without merging**